lib/cocoapods/validator.rb

```diff
@@ -560,12 +560,18 @@ def _validate_module_map
     # Ensures that a list of header files only contains header files.
     #
     def _validate_header_files(attr_name)
-      non_header_files = file_accessor.send(attr_name).
+      header_files = file_accessor.send(attr_name)
+      non_header_files = header_files.
         select { |f| !Sandbox::FileAccessor::HEADER_EXTENSIONS.include?(f.extname) }.
         map { |f| f.relative_path_from(file_accessor.root) }
       unless non_header_files.empty?
         error(attr_name, "The pattern matches non-header files (#{non_header_files.join(', ')}).")
       end
+      non_source_files = header_files - file_accessor.source_files
```

Comment:
I'm not completely sure, but I think that this screws up linting / uploading my Podspec.
Our source files are packaged in a .zip.  Our headers are in the .zip file (the zip can also be downloaded from our website).

Here's our spec:  https://github.com/CocoaPods/Specs/blob/master/Specs/CrittercismSDK/5.5.5/CrittercismSDK.podspec.json